### ISSUE TITLE
[Phase 7] Production Inference

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/seismic-event-detection#8]
## Phase 7: Production Inference

**Parent:** #1
**Depends on:** #7

### Objectives
Create production-ready inference class and demo script.

### Tasks

- [ ] Create `SeismicPicker` inference class
- [ ] Implement waveform preprocessing
- [ ] Add pick extraction logic
- [ ] Return structured predictions
- [ ] Create demo script
- [ ] Add sample waveform files
- [ ] Write usage documentation

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/inference.py` | Create | Production inference |
| `scripts/demo.py` | Create | Demo script |
| `examples/` | Create | Sample waveforms |

### Code Reference

```python
class SeismicPicker:
    """Production inference for seismic detection."""

    def __init__(self, model_path: str, device: str = 'auto'):
        self.device = 'cuda' if device == 'auto' and torch.cuda.is_available() else 'cpu'
        self.model = self._load_model(model_path)
        self.sample_rate = 100

    def predict(self, waveform: np.ndarray) -> dict:
        """Detect earthquake and pick P/S arrivals.

        Args:
            waveform: 3-component waveform (6000, 3)

        Returns:
            {
                'is_earthquake': bool,
                'detection_confidence': float,
                'p_arrival_sample': int,
                'p_arrival_time': float (seconds),
                's_arrival_sample': int,
                's_arrival_time': float (seconds),
            }
        """
        # Normalize
        waveform = (waveform - waveform.mean(axis=0)) / (waveform.std(axis=0) + 1e-8)

        # Predict
        with torch.no_grad():
            det, p, s = self.model(self._to_tensor(waveform))

        # Extract picks
        p_pick = self._extract_pick(p.squeeze().numpy())
        s_pick = self._extract_pick(s.squeeze().numpy())

        return {
            'is_earthquake': det.mean() > 0.5,
            'p_arrival_time': p_pick / self.sample_rate if p_pick >= 0 else None,
            's_arrival_time': s_pick / self.sample_rate if s_pick >= 0 else None,
        }
```

### Definition of Done

- [ ] Inference class loads model correctly
- [ ] Processes (6000, 3) waveforms
- [ ] Returns detection flag and pick times
- [ ] Demo script runs on sample data
- [ ] Error handling for invalid inputs
- [ ] Usage documented in README

### Technical Notes

**For junior developers:**
- Always normalize input waveforms
- Use torch.no_grad() for inference
- Return None for picks if confidence too low
- Sample rate is 100 Hz (convert samples to seconds)
